reservation/views.py

```diff
@@ -68,3 +42,127 @@ def get_csv_file(request):
         return response
 
     return render(request, 'reservation/user_reservation.html', {'reservations': reservations})
+
+
+@login_required()
+@ratelimit(key="ip", rate="30/m", block=True)
+def select_cinema(request):
+    cinemas = MovieTheater.objects.all()
+    return render(request, 'reservation/select_cinema.html', {'cinemas': cinemas})
+
+
+def select_movie(request, cinema_pk):
+    if request.method == "GET":
+        movies_playing_time = PlayingTime.objects.filter(assigned_hall__in=Hall.objects.filter(movie_theater=cinema_pk), start_time__range=get_current_week())
+        if movies_playing_time:
+            movies = [pt.assigned_movie for pt in movies_playing_time] # (pt -> playing_time)
+        else:
+            movies = []
+
+        return render(request, 'reservation/select_movie.html', {'movies': movies})
+    if request.method == "POST":
+        movie_pk = request.POST['movie_pk']
+        return redirect('select_playing_time', movie_pk=movie_pk, cinema_pk=cinema_pk)
+
+@login_required()
+@ratelimit(key="ip", rate="30/m", block=True)
+def select_playing_time(request, movie_pk, cinema_pk):
+    if request.method == "GET":
+        playing_times = PlayingTime.objects.filter(
+            assigned_movie=movie_pk,
+            assigned_hall__in=Hall.objects.filter(movie_theater=cinema_pk),
+        )
+        return render(
+            request, "reservation/select_playing_time.html", {"playing_times": playing_times}
+        )
+
+@login_required()
+@ratelimit(key="ip", rate="30/m", block=True)
+def select_seats(request, pk):
+    if request.method == "GET":
+        reservations = Reservation.objects.filter(
+            playing_time=PlayingTime.objects.get(pk=pk)
+        )
+        reserved_seats = [item.seat.seat_name for item in reservations]
+        seats = Seat.objects.filter(halls=PlayingTime.objects.get(pk=pk).assigned_hall.id).order_by('seat_name')
```

Comment:
Even if you don't want to make the above changes, it's worth saving the value of PlayingTime.objects.get(pk=pk) into a variable, to avoid querying the same object multiple times.